Clarified a description about auto-incremental ID (#7811)
pingcap/docs: mysql-compatibility.md

### Auto-increment ID

> + In TiDB, auto-incremental columns are globally unique, but only guaranteed to be incremental and sequential on a single TiDB server. They are *not* guaranteed to be incremental among multiple TiDB servers or allocated sequentially. It is recommended that you do not mix default values and custom values. Otherwise, you might encounter the `Duplicated Error` error message.

Reviewer: ```suggestion
+ In TiDB, auto-incremental columns are globally unique and incremental on a single TiDB server. They are *not* necessarily incremental and sequential among multiple TiDB servers. It is recommended that you do not mix default values and custom values. Otherwise, you might encounter the `Duplicated Error` error message.
```
原文是 “TiDB 的自增列仅保证唯一，也能保证在单个 TiDB server 中自增，但不保证多个 TiDB server 中自增，不保证自动分配的值的连续性”。

Reviewer 2: @shichun-0415 The original Chinese version also needs to be updated. See https://github.com/pingcap/docs-cn/pull/8604/files.

> + In TiDB, auto-incremental columns are globally unique, but only guaranteed to be incremental and sequential on a single TiDB server. They are *not* guaranteed to be incremental among multiple TiDB servers or allocated sequentially. It is recommended that you do not mix default values and custom values. Otherwise, you might encounter the `Duplicated Error` error message.

Reviewer 2: How about this version? We also have one FAQ about this in https://docs.pingcap.com/tidb/stable/sql-faq#why-does-the-auto-increment-id-of-the-later-inserted-data-is-smaller-than-that-of-the-earlier-inserted-data-in-tidb ([Chinese version](https://docs.pingcap.com/zh/tidb/stable/sql-faq#tidb-%E4%B8%AD%E4%B8%BA%E4%BB%80%E4%B9%88%E5%87%BA%E7%8E%B0%E5%90%8E%E6%8F%92%E5%85%A5%E6%95%B0%E6%8D%AE%E7%9A%84%E8%87%AA%E5%A2%9E-id-%E5%8F%8D%E8%80%8C%E5%B0%8F))
```suggestion
+ In TiDB, auto-incremental columns are globally unique. They are incremental on a single TiDB server, but *not* necessarily incremental among multiple TiDB servers or allocated sequentially. It is recommended that you do not mix default values and custom values. Otherwise, you might encounter the `Duplicated Error` error message.
```
